frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,206 @@
+      _.each(
+        this.state.items,
+        (props, key) =>
+          (items[key] = key ? <StorageClassDropdownEntry {...props} /> : <StorageClassDropdownNoStorageClassOption {...props} />)
+      );
+
+      const { selectedKey, defaultClass } = this.state;
+
+      const Component = <Dropdown
+        autocompleteFilter={this.autocompleteFilter}
+        autocompletePlaceholder={placeholder}
+        items={items}
+        selectedKey={selectedKey}
+        title={this.state.title}
+        onChange={this.onChange}
+        id={id}
+        menuClassName="dropdown-menu--text-wrap"
+      />;
+      // Only show the dropdown if 'no storage class' is not the only option which depends on defaultClass
+      const itemsAvailableToShow = (_.size(items)>1 && !defaultClass) || (_.size(items)>0 && defaultClass);
+      const helpMes = defaultClass ? `Default storage class ${defaultClass} will be used if no other storage class selected.` : 'Optional storage class for the new claim.';
+      return <React.Fragment>
+        {loaded && itemsAvailableToShow &&
+          <div>
+            <label className="control-label" htmlFor="storageclass-dropdown">
+            Storage Class
+            </label>
+            {Component}
+            <p className="help-block">
+              {helpMes}
+            </p>
+          </div>
+        }
+      </React.Fragment>;
+    }
+}
+
+export const StorageClassDropdown = props => {
+  return <Firehose resources={[{ kind: 'StorageClass', prop: 'StorageClass', isList: true }]}>
+    <StorageClassDropdown_ {...props} />
+  </Firehose>;
+};
+
+const StorageClassDropdownEntry = props => {
+  const storageClassProperties = [props.default ? ' (default)' : '', props.description, props.accessMode, props.provisioner, props.type, props.zone];
+  const storageClassDescriptionLine = storageClassProperties.filter(Boolean).join(' | ');
+  return <div className="co-storage-class-dropdown-flex-column">
+    <div className="co-storage-class-dropdown-flex-row">
+      <span className="co-storage-class-dropdown-icon-column"><ResourceIcon kind={props.kindLabel} /></span> <span>{props.name}</span>
+    </div>
+    <div className="co-storage-class-dropdown-flex-row">
+      <div className="co-storage-class-dropdown-icon-column"> &nbsp;</div><div className="text-muted"> {storageClassDescriptionLine}</div>
+    </div></div>;
+};
+
+const StorageClassDropdownNoStorageClassOption = props => {
+  return <div className="co-storage-class-dropdown-flex-column">
+    <div className="co-storage-class-dropdown-flex-row">
+      <span className="co-storage-class-dropdown-icon-column"></span> <span>{props.name}</span>
+    </div></div>;
+};
+
+export type StorageClassDropdownState = {
+    items: any;
+    name: string;
+    selectedKey: string;
+    title: any;
+    defaultClass: string;
+};
+
+export type StorageClassDropdownProps = {
+    id?: string;
+    loaded?: boolean;
+    name: string;
+    onChange: any;
+    placeholder?: string;
```

Comment:
Does this need to be configurable? I would just hardcode a placeholder until we have a need to customize it.